packages/itwin/grouping-mapping-widget/src/components/Mappings/hooks/useMappingsOperations.ts

```diff
@@ -91,63 +17,54 @@ export interface MappingsOperationsProps extends GroupingMappingApiConfig {
-  }, [extractionStatusIcon]);
+  const {
+    data: extractionStatus,
+    isFetching: isLoadingExtractionStatus,
+  } = useFetchExtractionStatus({ iModelId, getAccessToken, extractionClient });
 
-  useEffect(() => {
-    if(previousStatusIcon.current?.iconStatus === undefined) {
-      void fetchExtractionStatus(
-        iModelId,
-        getAccessToken,
-        extractionClient,
-        setExtractionStatusIcon,
-        setExtractionMessageData
-      );
-    }
-  }, [iModelId, getAccessToken, extractionClient, isLoading, setExtractionStatusIcon, setExtractionMessageData]);
+  const refreshExtractionStatus = useCallback(async () => {
+    await queryClient.invalidateQueries({queryKey: ["extractionStatus"]});
+  }, [queryClient]);
 
-  const refresh = useCallback(async () => {
-    setMappings([]);
-    await Promise.all([
-      fetchExtractionStatus(iModelId, getAccessToken, extractionClient, setExtractionStatusIcon, setExtractionMessageData),
-      fetchMappings(setMappings, iModelId, setIsLoading, getAccessToken, mappingClient, setErrorMessage),
-    ]);
-  }, [extractionClient, getAccessToken, mappingClient, iModelId, setExtractionMessageData, setExtractionStatusIcon]);
+  const refreshMappings = useCallback(async () => {
+    await queryClient.invalidateQueries({queryKey: ["mappings"]});
+  }, [queryClient]);
 
-  const toggleExtraction = useCallback(async (mapping: Mapping) => {
-    try {
-      setIsTogglingExtraction(true);
+  const { mutateAsync: toggleExtraction, isLoading: isTogglingExtraction } = useMutation({
+    mutationFn: async (mapping: Mapping) => {
+      const accessToken = await getAccessToken();
       const newState = !mapping.extractionEnabled;
+      await mappingClient.updateMapping(accessToken, iModelId, mapping.id, { extractionEnabled: newState });
+    },
+    onSuccess: async () => {
+      await refreshMappings();
+    },
+  });
+
+  const { mutateAsync: onDelete, isLoading: isDeletingMapping} = useMutation({
+    mutationFn: async (mapping: Mapping) => {
       const accessToken = await getAccessToken();
-      await mappingClient.updateMapping(accessToken, iModelId, mapping.id, {
-        extractionEnabled: newState,
-      });
-    } catch (error: any) {
-      setErrorMessage(getErrorMessage(error.status));
-    } finally {
-      setIsTogglingExtraction(false);
-    }
-  }, [getAccessToken, iModelId, mappingClient]);
+      await mappingClient.deleteMapping(accessToken, iModelId, mapping.id);
+    },
+    onSuccess: async () => {
+      await refreshMappings();
+    },
+  });
 
-  const onDelete = async (mapping: Mapping) => {
-    const accessToken = await getAccessToken();
-    await mappingClient.deleteMapping(accessToken, iModelId, mapping.id);
-    await refresh();
-  };
+  const isLoading = isLoadingMappings || isLoadingExtractionStatus || isTogglingExtraction || isDeletingMapping;
+  const extractionStatusGated = extractionStatus ??  {extractionStatusIcon: {
```

Comment:
Looks like you have a double space there after `??`.

Reply:
Fixed.